### PR TITLE
[#12588] Added unit tests for CommentTableModalComponent

### PR DESCRIPTION
**Pull request by kenneySiu**
Part of #12588

**Outline of Solution**
Added unit tests for the following methods in CommentTableModalComponent
ngOnChanges,
triggerDeleteCommentEvent,
triggerUpdateCommentEvent,
triggerSaveNewCommentEvent,
triggerModelChange

### CODE REVIEW

File: src/web/app/components/comment-box/comment-table-modal/comment-table-modal.component.spec.ts [hunk 1]
```diff
@@ -52,4 +54,68 @@ describe('CommentTableModalComponent', () => {
...
+          commentText: '',
+          isUsingCustomVisibilities: false,
+          showCommentTo: [],
+          showGiverNameTo: [],
+        },
+        isEditing: true,
+      },
+      isAddingNewComment: false,
+      isReadOnly: false,
+    };
```
Comment (cedricongjh) [inline src/web/app/components/comment-box/comment-table-modal/comment-table-modal.component.spec.ts:73]: let's abstract this out for multiple tests to use

File: src/web/app/components/comment-box/comment-table-modal/comment-table-modal.component.spec.ts [hunk 2]
```diff
@@ -52,4 +54,68 @@ describe('CommentTableModalComponent', () => {
...
+    component.model = testModel;
+    component.ngOnChanges();
+    expect(ngOnChangesSpy).toHaveBeenCalledWith({
+      ...testModel,
+      isAddingNewComment: true,
+    });
+  });
+
+  it('should trigger an event to delete comment from comments table', () => {
```
Comment (cedricongjh) [inline src/web/app/components/comment-box/comment-table-modal/comment-table-modal.component.spec.ts:82]:
```suggestion
  it('should emit a DeleteCommentEvent with the correct index when triggerDeleteCommentEvent is called', () => {
```

let's follow this format when describing tests for event emissions, do the same for the other tests that emit events

**Review by cedricongjh — CHANGES_REQUESTED** [review]
thank you for the PR, just a few minor nits!

**Comment by kenneySiu** [comment]
Ready for review

**Pull request merged**